src/plugins/search/providers/ngv-osm-provider.ts

```diff
@@ -0,0 +1,45 @@
+import type {Geometry} from 'geojson';
+import type {
+  INGVOsmSearchProviderConfig,
+  INGVSearchProvider,
+  INGVSearchResult,
+} from '../../../interfaces/search/ingv-search-provider.js';
+
+const nominatimSearchUrl =
+  'https://nominatim.openstreetmap.org/search?q={input}&accept-language={lang}&limit={limit}&format=jsonv2';
+
+interface OSMFeature {
+  display_name: string;
+  lat: string;
+  lon: string;
+}
+
+export class NGVOsmSearchProvider implements INGVSearchProvider {
+  public name: 'osm';
```

Comment:
Better call it nominatim, because there are many possible providers based on OSM data.